tests/nnstreamer_filter_extensions_common/unittest_tizen_template.cpp

```diff
@@ -0,0 +1,264 @@
+ */
+TEST (nnstreamer_EXT_NICK_NAME_basic_functions, open_close_01_n)
+{
+  int ret;
+  void *data = NULL;
+  gchar **model_files;
+
+  model_files = get_model_files ();
+  ASSERT_TRUE (model_files != nullptr);
+
+  GstTensorFilterProperties prop = {
+    .fwname = "EXT_NAME",
+    .fw_opened = 0,
+    .model_files = const_cast<const char **>(model_files),
+    .num_models = (int) g_strv_length (model_files),
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("EXT_NAME");
+  EXPECT_NE (sp, (void *) NULL);
+
+  /** close without open, should not crash */
+  sp->close (&prop, &data);
+
+  /** open and close successfully */
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+  sp->close (&prop, &data);
+
+  /** close twice, should not crash */
+  sp->close (&prop, &data);
+  g_strfreev (model_files);
+}
+
+/**
+ * @brief Get input/output dimensions with EXT_NICK_NAME subplugin
+ */
+TEST (nnstreamer_EXT_NICK_NAME_basic_functions, get_dimension)
+{
+  int ret;
+  void *data = NULL;
+  GstTensorsInfo res;
+  gchar **model_files;
+
+  model_files = get_model_files ();
+  ASSERT_TRUE (model_files != nullptr);
+
+  GstTensorFilterProperties prop = {
+    .fwname = "EXT_NAME",
+    .fw_opened = 0,
+    .model_files = const_cast<const char **>(model_files),
+    .num_models = (int) g_strv_length (model_files),
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("EXT_NAME");
+  EXPECT_NE (sp, (void *) NULL);
+
+  /** get input/output dimension without open */
+  ret = sp->getInputDimension (&prop, &data, &res);
+  EXPECT_NE (ret, 0);
+  ret = sp->getOutputDimension (&prop, &data, &res);
+  EXPECT_NE (ret, 0);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+
+  /** get input/output dimension successfully */
+  ret = sp->getInputDimension (&prop, &data, NULL);
+  EXPECT_NE (ret, 0);
+  ret = sp->getInputDimension (&prop, &data, &res);
+  EXPECT_EQ (ret, 0);
+
+  ret = sp->getOutputDimension (&prop, &data, NULL);
+  EXPECT_NE (ret, 0);
+  ret = sp->getOutputDimension (&prop, &data, &res);
+  EXPECT_EQ (ret, 0);
+
+  sp->close (&prop, &data);
+  g_strfreev (model_files);
+}
+
+/**
+ * @brief Test EXT_NICK_NAME subplugin with successful invoke
+ */
+TEST (nnstreamer_EXT_NICK_NAME_basic_functions, invoke)
+{
+  int ret;
+  void *data = NULL;
+  GstTensorMemory input, output;
+  gchar **model_files;
+  GstTensorsInfo res;
+  int num_inputs, num_outputs;
+
+  model_files = get_model_files ();
+  ASSERT_TRUE (model_files != nullptr);
+
+  GstTensorFilterProperties prop = {
+    .fwname = "EXT_NAME",
+    .fw_opened = 0,
+    .model_files = const_cast<const char **>(model_files),
+    .num_models = (int) g_strv_length (model_files),
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("EXT_NAME");
+  EXPECT_NE (sp, (void *) NULL);
+
+  /** invoke without open */
+  ret = sp->invoke_NN (&prop, &data, &input, &output);
+  EXPECT_NE (ret, 0);
```

Comment:
Let's make sure these have ```_n``` postfixes in their test case names.

Reply:
updated - split test cases into successful and negative ones (negative ones suffixed with `_n`)